text-feature-translation/src/main/java/net/kyori/adventure/text/feature/translation/TranslationRegistry.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2020 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.text.feature.translation;
+
+import net.kyori.adventure.text.renderer.ComponentRenderer;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import java.util.Locale;
+
+/**
+ * A registry of translators for rendering components.
+ */
+public interface TranslationRegistry extends Translator, ComponentRenderer<Locale> {
+  /**
+   * Gets the translation registry.
+   *
+   * @return the translation registry
+   */
+  static TranslationRegistry get() {
```

Comment:
I don't know if I like the idea of global stored state -- this seems like an easy way to accidentally end up with message conflicts.

Reply:
Agreed. I think it should be the platform's job to provide the translation registry for plugins.

Reply:
I would agree platforms should handle the translation rendering layer, but I think having a platform-neutral entrypoint for keys is the right way to go. Keys are typically neutral of platform.